packages/terra-date-picker/src/react-datepicker/calendar.jsx

```diff
@@ -277,14 +277,14 @@ export default class Calendar extends React.Component {
     }
   }
 
-  componentWillReceiveProps (nextProps) {
-    if (nextProps.preSelection && !isSameDay(nextProps.preSelection, this.props.preSelection)) {
+  componentDidUpdate (prevProps) {
+    if (this.props.preSelection && !isSameDay(this.props.preSelection, prevProps.preSelection)) {
       this.setState({
-        date: this.localizeDate(nextProps.preSelection)
+        date: this.localizeDate(this.props.preSelection)
       })
-    } else if (nextProps.openToDate && !isSameDay(nextProps.openToDate, this.props.openToDate)) {
+    } else if (this.props.openToDate && !isSameDay(this.props.openToDate, nextProps.openToDate)) {
```

Comment:
prevProps instead of nextProps

Reply:
Updated here https://github.com/cerner/terra-framework/pull/877/commits/350d1e0ab77f4c65d356bba824a401c61e8dcc53.